src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -49,67 +81,146 @@ public String path() {
                     try (OutputStream response = exchange.getResponseBody()) {
+                        List<SerializableRecordingDescriptor> recordings = getRecordings();
+                        exchange.sendResponseHeaders(HttpStatus.SC_OK, 0);
                         mapper.writeValue(response, recordings);
+                    } catch (Exception e) {
+                        log.error("recordings serialization failure", e);
+                    }
+                    break;
+                case "POST":
+                    try (InputStream body = exchange.getRequestBody()) {
+                        StartRecordingRequest req =
+                                mapper.readValue(body, StartRecordingRequest.class);
+                        if (!req.isValid()) {
+                            exchange.sendResponseHeaders(HttpStatus.SC_BAD_REQUEST, -1);
+                            return;
+                        }
+                        SerializableRecordingDescriptor recording = startRecording(req);
+                        exchange.sendResponseHeaders(HttpStatus.SC_CREATED, 0);
+                        try (OutputStream response = exchange.getResponseBody()) {
+                            mapper.writeValue(response, recording);
+                        }
+                    } catch (QuantityConversionException
+                            | ServiceNotAvailableException
+                            | FlightRecorderException
+                            | org.openjdk.jmc.rjmx.services.jfr.FlightRecorderException
+                            | InvalidEventTemplateException
+                            | InvalidXmlException
+                            | IOException e) {
+                        log.error("Failed to start recording", e);
+                        exchange.sendResponseHeaders(HttpStatus.SC_INTERNAL_SERVER_ERROR, -1);
                     }
-                } catch (Exception e) {
-                    log.error("recordings serialization failure", e);
-                } finally {
-                    exchange.close();
-                }
-                break;
-            default:
-                exchange.sendResponseHeaders(HttpStatus.SC_NOT_FOUND, -1);
-                exchange.close();
-                break;
+                    break;
+                default:
+                    log.warn("Unknown request method {}", mtd);
+                    exchange.sendResponseHeaders(HttpStatus.SC_METHOD_NOT_ALLOWED, -1);
+                    break;
+            }
+        } finally {
+            exchange.close();
+        }
+    }
+
+    private boolean ensureMethodAccepted(HttpExchange exchange) throws IOException {
+        Set<String> blocked = Set.of("POST");
+        String mtd = exchange.getRequestMethod();
+        boolean restricted = blocked.contains(mtd);
+        if (!restricted) {
+            return true;
+        }
+        boolean passed = restricted && MutatingRemoteContext.apiWritesEnabled(config);
```

Comment:
Should this `restricted` check be removed here since we already return early if false?